tools/wt_verify.py

```diff
@@ -38,24 +38,127 @@
 import os
 import subprocess
 import sys
+import json
+import matplotlib.pyplot as plt
+import re
 
-def visualize(data, visualization_type):
+SEPARATOR = "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n"
+WT_OUTPUT_FILE = "wt_output_file.txt"
+
+def output_pretty(output):
     """
-    Visualizing data
+    Print output for parsing script
     """
-    pass
+    str = "{"
+    for i, checkpoint in enumerate(output):
+        if i:
+            str += ","
+        str += "\n\t" + checkpoint + ": {"
+        for j, key in enumerate(output[checkpoint]):
+            value = output[checkpoint][key]
+            if j:
+                str += ","
+            str += "\n\t\t" + key + ": {\n\t\t\t" + json.dumps(value)
+            str += "\n\t\t}"
+        str += "\n\t}"
+    str += "\n}"
+    return str
+
+
+def is_int(string):
+    """
+    Check if string can be converted to an integer
+    """
+    if isinstance(string, str):
+        try :
+            return int(string)
+        except ValueError:
+            return string
+    return string
+
+
+def string_to_iterable(line):
+    """
+    Helper that converts string to dictionaries and / or lists
+    """
+    dict = {}
+    for x in line.split(" | "):
+        [key, value] = x.split(": ", 1)
+        if value[0] == "[" and value[-1] == "]":
+            value = value[1:-1].split(", ")
+            value = list(map(lambda n: is_int(n), value))
+        if key == "addr": 
+            temp = value[0].split(": ")
+            dict.update({"object_id": is_int(temp[0]), "offset_range": temp[1], "size": is_int(value[1]), 
+                         "checksum": is_int(value[2])})
+        else:
+            dict[key] = is_int(value)
+    return dict
+
+
+def parse_node(f, line, output, chkpt_info, root_addr, is_root_node):
+    node = {}
+    line = line[2:-1] # remove new node symbol
+    page_type = line.split()[-1]
+    assert page_type == "internal" or page_type == "leaf"
+    node_id = is_int(line.split(": ")[0])
+    line = f.readline()
+    while line and line != SEPARATOR and not line.startswith("- "):
+        if line.startswith("\t> "): 
+            node.update(string_to_iterable(line[3:-1])) # remove metadata symbol
```

Comment:
could we name this parse_metadata and have line 107 in the function itself?

Reply:
fixed https://github.com/wiredtiger/wiredtiger/pull/10164/commits/052ab3b4c85bedfb931aea4e92262e2fd24a4086